src/utils.spec.tsx

```diff
@@ -188,4 +195,46 @@ describe(__filename, () => {
       expect(getPathFromQueryString(history)).toEqual(null);
     });
   });
+
+  describe('createCodeLineAnchorGetter', () => {
+    it('returns getCodeLineAnchor if compareInfo is null', () => {
+      expect(createCodeLineAnchorGetter(null)).toEqual(getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor if compareInfo is undefined', () => {
+      expect(createCodeLineAnchorGetter(undefined)).toEqual(getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor if compareInfo.diff is falsey', () => {
+      expect(
+        createCodeLineAnchorGetter({ diff: null, mimeType: 'mime/type' }),
+      ).toEqual(getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor from ForwardComparisonMap is compareInfo.diff exists', () => {
+      const compareInfo = createFakeCompareInfo();
+      if (!compareInfo.diff) {
+        throw new Error('compareInfo.diff was unexpectedly empty');
+      }
+      const map = new ForwardComparisonMap(compareInfo.diff);
+
+      const getterFromFactory = createCodeLineAnchorGetter(compareInfo);
+      const getterFromMap = map.createCodeLineAnchorGetter();
+
+      // Generate a getter without compareInfo to verify that it returns a
+      // different result.
+      const getterFromFactoryWithoutCompareInfo = createCodeLineAnchorGetter(
+        null,
+      );
+
+      // We cannot use `toEqual` for the two getters above because they are
+      // different instances, so we check that they return the same thing.
+      expect(getterFromFactory(1)).toEqual(getterFromMap(1));
```

Comment:
I also tried to create a manual mock of the class object but couldn't figure it out. Oh, well. Your approach here seems fine.